resources/roca_lib.brs

```diff
@@ -123,16 +125,22 @@ end function
 ' Creates a test case with a given description.
 ' @param description a string describing this test case
 ' @param func the function to execute as part of this test case
-sub __it(description as string, func as object)
-    m.__suite.__registerCase("default", description, m.__suite, func)
+sub __it(description as string, func as object, args = invalid as dynamic)
```

Comment:
Any particular reason we're exposing `args` as a new parameter to `m.it` here?  Now that we're checking for `m.func_args <> invalid` on line 270 (for the `m.it_each` use-case) I'm not sure I understand why this param is necessary 😃

Reply:
1. I don't wanna another one copy of `m.it` logic but with some small changes - like passing args to m.it
2. So `m.it_each` will be just wrapper that expands to several m.it with passing args from an array of args
3. Also, it will add a new feature to `m.it` - passing arg manually, eg. when m.it is not too good for using them in `m.it_each`, cuz they use same data but in a different way(one `it` checks if a passed number is positive, and next `it` checks if that number is odd)

I think it is a little bit stupid to pass a standalone `invalid` as an argument. So it is pretty ok to use it as a detector whether `func` has parameters or it is empty. 
Another way is using  `args = uninitialized as dynamic ` and then use next check `if type(m.func_args) <> "uninitialized" `

Reply:
> I think it is a little bit stupid to pass a standalone `invalid` as an argument. So it is pretty ok to use it as a detector whether `func` has parameters or it is empty.

That's fair!  We could avoid that by changing `__suite_registerCase` to accept an optional `args = invalid as dynamic`, allowing `m.it()` and `m.it_each` to both directly call `m.__suite.__registerCase` I suppose. 

I don't mind having the extra argument around, though I do wonder whether we'll need the first position of each callback in the future.  Oh well, we'll deal with that when we get there 😄